src/pages/ProductDetail/ProductDetail.js

```diff
@@ -1,9 +1,270 @@
+                style={{
+                  transform: `translate3d(
+                ${imageNum}px, 0px, 0px`,
+                }}
+              >
+                {productInfoList.images &&
+                  productInfoList.images.map(image => {
+                    return (
+                      <span className="slideContent" key="image.id">
+                        <ProductPhoto imageUrl={image.image_url} />
+                      </span>
+                    );
+                  })}
+              </div>
+            </div>
+            <div className="productText">
+              <div className="productName">
+                {productInfoList && productInfoList.name}
+              </div>
+              <div className="productPrice">
+                <div className="productColor">
+                  {productInfoList.colors && productInfoList.colors[0].name}
+                  <span className="productColorDivision">•</span>
+                  {productInfoList.colors && productInfoList.colors[0].name}
+                </div>
+                <div>
+                  {productInfoList.price &&
+                    productInfoList.price
+                      .toString()
+                      .replace(/\B(?=(\d{3})+(?!\d))/g, ',')}
+                  원
+                </div>
+              </div>
+              <div
+                className="bonus"
+                onClick={() => {
+                  this.setState({ isBonus: false });
+                }}
+              >
+                카드 혜택보기
+              </div>
+              <div className="colorNumber">
+                {productInfoList.colors && productInfoList.colors.length} 개의
+                색상
+              </div>
+              <div className="colorBox">
+                {productInfoList.colors &&
+                  productInfoList.colors.map(color => {
+                    return (
+                      <div className="outCircle" key={color.id}>
+                        <div
+                          className={`color${color} circle`}
+                          style={{
+                            backgroundColor: color.name,
+                          }}
+                        ></div>
+                      </div>
+                    );
+                  })}
+              </div>
+              <div
+                className="productSize"
+                onClick={() => {
+                  this.setState({ isSize: false });
+                }}
+              >
+                <button>
+                  SIZE &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{size}
+                  <i className="fas fa-chevron-right"></i>
+                </button>
+              </div>
+              <div
+                className={size ? 'productPickOff' : 'productPickOn'}
+                onClick={() => {
+                  size
+                    ? this.setState({ isShoppingBag: false })
+                    : this.setState({ isSize: false });
+                }}
```

Comment:
```suggestion
                onClick={() => {
                    this.setState({ [size ? 'isShoppingBag' : 'isSize']: false })
                }}
```
이렇게 하면 동일하긴 할텐데 애초에 위에서 정의해둔 함수들을 사용하지 못하는게 조금 걸리네요 이건 조금 생각해보죠